types/mui-datatables/index.d.ts

```diff
@@ -103,12 +103,18 @@ export interface MUIDataTableTextLabels {
     selectedRows: MUIDataTableTextLabelsSelectedRows;
 }
 
+export interface MUIDataTableFilterOptions {
+    names?: string[];
+    display?: (filterList: string[], onChange: any, index: number, column: any) => void;
+    logic?: any;
```

Comment:
Took a closer look at how they use the logic prop in [this example](https://github.com/gregnb/mui-datatables/blob/master/examples/customize-filter/index.js) and I think it can be typed as such.

```ts
logic?: (prop: string, filterValue: any[]) => boolean
```

Reply:
Tried it and it compiles and runs fine. Added to PR.